cfme/middleware/jdr_report.py

```diff
@@ -0,0 +1,142 @@
+import re
+from datetime import datetime
+
+from cfme.utils import attributize_string
+from cfme.utils.appliance import current_appliance
+
+
+def _db_select_query(server):
+    """column order: `id`, `queued_at`, `requesting_user`, `status`"""
+    t_ms = current_appliance.db.client['middleware_servers']
+    t_mdr = current_appliance.db.client['middleware_diagnostic_reports']
+    query = current_appliance.db.client.session.query(
+        t_mdr.id,
+        t_mdr.queued_at,
+        t_mdr.requesting_user.label('username'),
+        t_mdr.status).join(t_mdr, t_mdr.middleware_server_id == t_ms.id)
+    if server:
+        query = query.filter(t_ms.name == server.name)
+    return query
+
+
+class JDRReport():
+    """
+    JDRReport class provides actions and details of JDR Reports List
+     on Middleware Server page.
+    Class method available to get existing JDR list for provided Server
+
+    Args:
+        username: user queued the JDR Report
+        status: status of the JDR Report
+        server: Server for which report is generated
+        filename: name of the JDR Report file
+        size: size of the JDR Report file
+        queued_at: queued time of the JDR Report
+        db_id: database row id of JDR Report
+
+    Usage:
+
+    .. code-block:: python
+
+        myreport = JDRReport(username='admin', status='Ready', server=eap_server)
+        myreport.download()
+
+        myreports = JDRReport.reports()
+
+    """
+
+    def __init__(self, username, status, server, appliance=None, **kwargs):
+        self.username = username
+        self.status = status
+        self.server = server
+        self.filename = kwargs.get('filename')
+        self.size = kwargs.get('size')
+        self.queued_at = kwargs.get('queued_at')
+        self.db_id = kwargs.get('db_id')
+        if 'properties' in kwargs:
+            for prop in kwargs['properties']:
+                # check the properties first, so it will not overwrite core attributes
+                if getattr(self, attributize_string(prop), None) is None:
+                    setattr(self, attributize_string(prop), kwargs['properties'][prop])
+
+    @classmethod
+    def reports(cls, server):
```

Comment:
This is exactly what are collections good for .... no more classmethods with passing parents around :) These classmethods would live in the collection.